cfg/yamllint-config.yaml

```diff
@@ -0,0 +1,39 @@
+---
+extends: 'default'
+rules:
+  braces:
+    forbid: true
+  brackets:
+    forbid: true
+  comments:
+    ignore-shebangs: true
+    level: 'error'
+    min-spaces-from-content: 1
+    require-starting-space: true
+  comments-indentation:
+    level: 'error'
+  document-end: 'enable'
+  document-start:
+    level: 'error'
+  empty-lines:
+    max: 0
+    max-end: 1
+    max-start: 0
+  empty-values: 'enable'
+  float-values:
+    forbid-inf: false
+    forbid-nan: false
+    forbid-scientific-notation: true
+    require-numeral-before-decimal: true
+  indentation:
+    check-multi-line-strings: true
+    indent-sequences: false
+    spaces: 2
+  key-ordering: 'enable'
```

Comment:
So I actually got used to and like most of the configurations for the yamllint although the alphabetical ordering one often moves around the keys to non-standard orders for things like `.github/actions/*.yaml` files.

Like for example, normally the `runs-on` key is typically used before `steps` because it's a more intuitive logical flow. Same with `name` at the top, rather than near the middle

Any way we could not have this enabled? I imagine the cloud infra will be pretty heavy on yaml files and we'll have to re-order a lot of stuff if we keep it

Reply:
I agree here that there is certainly a trade-off in terms of intuitive appeal versus consistency. In the context for actions we can at least know that there should be a `runs-on`, so even if it is out of order it is still there, but yes this diverges from the common templates

I suppose I'm just more concerned about optimizing out the other end of the spectrum, e.g. where we have a giant CloudFormation stack file and we don't know where to look because "logical" ordering isn't necessarily intuitive and then we end up sifting through pages of unalphabetized declarations

Can always revisit later, but for now perhaps we can just rely on escapes in situations where it might be confusing